src/Features/CSharp/Portable/Completion/SuggestionMode/CSharpSuggestionModeCompletionProvider.cs

```diff
@@ -211,23 +211,25 @@ private bool IsPotentialPatternVariableDeclaration(SyntaxToken token)
                 return false;
             }
 
-            // Statements containing 'not' cannot be valid variable declarations, e.g. 'e is not int $$' and 'e is not (1 and int $$)'
-            if (patternSyntax.GetAncestorsOrThis(a => a is UnaryPatternSyntax unaryPatternSyntax &&
-                                                      unaryPatternSyntax.PatternOperator.IsKind(SyntaxKind.NotKeyword)).Any())
+            var possibleInvalidVariable = patternSyntax.Parent;
+            while (possibleInvalidVariable is BinaryPatternSyntax ||
+                   possibleInvalidVariable is UnaryPatternSyntax ||
+                   possibleInvalidVariable is ParenthesizedPatternSyntax ||
+                   possibleInvalidVariable is ParenthesizedExpressionSyntax)
```

Comment:
i think you want to walk up any pattern parent.  imagine you had:

```e is int or { } $$```

this can't have a name here since it's in an or.  but hte parent pattern woudl be a recursive pattern.  absically, you just want to see if you're *anywhere* underneath an or/not pattern.

Reply:
should never have parenthesized expr here afaict.  can you give a case/test where that happens.